### ISSUE TITLE
[Feature Requests] :Yanks menu and last paste memory options

### ISSUE THREAD

**Issue by fowlslegs** [svermeulen/vim-easyclip#54]
It would be nice if after pulling up the list :Yanks<CR> I could press j,k to move up and down among the entries list and press enter to paste the selected on where the cursor is. It would be even nicer if I could press 12<CR> and the 12th item in the register would be selected. Another feature the yanks list could have would be allowing a chord like 12m1 which would move the 12th register to the 1st, the 1st to the 2nd, 2nd to the 3rd, and so on.

Another boolean option that would be nice to have is the ability to set the last paste as the first register. So if I press p<c-p><c-p> that would paste the 3rd register. Then as soon as I press a motion or insert key (or otherwise lose the ability to continue using <c-p> and <c-n> to cycle through keys) easyclip would move the 3rd register to to the 1st, the 1st to the 2nd, and the 2nd to the 3rd.

Great work on the clipboard btw. My favorite I've found so far because of the simplicity and handy substitution motions, but lacks a few features YankRing and others have.

**Comment by svermeulen** [comment]
Yeah, I agree, it would be cool if the `:Yanks` command would open a scratch buffer that you could navigate within.  I'm not sure I understand your second suggestion of the boolean option.  Can you elaborate on that?

Thanks - glad you are enjoying it.  Out of curiosity, what other features are you referring to that YankRing and others have?

**Comment by fowlslegs** [comment]
I didn't escape the \<C-P>s in my sentence "So if I press p\<C-P>\<C-P> that would paster the 3rd register." Hopefully it makes sense when you read it now. Essentially, if you use the text in register 3 it would move the contents of that register to register 1 and push the old 1 and 2 back. Further, you could add an easy "escape" so that this will not happen (or vice-versa when the boolean option is not chosen). Perhaps this would better explain it:

* **g:LastPasteToFront=0:** This would normally behave like easyclip currently does. However, say I press p<C-p><C-p> or "3p. If I then immediately afterwards press a key, e.g., \<C-F> it will move the contents of register 3 to register 1 as described above.

* **g:LastPasteToFront=1:** This option would move the last pasted register to position 1 by default. If I press \<C-F> immediately after "3p in this setting, it would instead *not* move register 3 to register 1, i.e., all register positions would be preserved.

Some other comments:
* I think it would be good to map M to m$ by default as that follows the convention of D being equivalent to d$.
* I often use xp to transpose two letters I've typed. It might be nice to save 1 character deletions made with x for a brief second, then delete them if the next key that isn't pressed is p.

I think the defaults for easyclip are very easy to understand, and that you can add more functionality that people don't need to use. Extra features that are there if you want them, but don't need to be learned in order to operate a pretty good clipboard.

**Comment by svermeulen** [comment]
Just wanted to mention here that a possible solution to this was posted at #62
